src/repo/NxRepositoryParser.ts

```diff
@@ -0,0 +1,163 @@
+import { JSONSchemaForNPMPackageJsonFiles } from "@schemastore/package";
+import fs, { readFileSync } from "fs";
+import _ from "lodash";
+import path from "path";
+import { gt } from "semver";
+import util from "util";
+import vscode from "vscode";
+import { Log } from "vscode-test-adapter-util";
+import { EXPERIMENTAL_NX_CLI_FEATURE_TOGGLE, EXTENSION_CONFIGURATION_NAME } from "../constants";
+import RepoParserBase from "./RepoParserBase";
+import { JestExecutionParams, ProjectConfig, RepoParser } from "./types";
+
+// the following requires Node 8 minimum.
+export const exists = util.promisify(fs.exists);
+export const readFile = util.promisify(fs.readFile);
+
+interface NxJestOptions {
+  jestConfig: string;
+  tsConfig?: string;
+  setupFile?: string;
+}
+
+interface NxProject {
+  projectType: string;
+  sourceRoot: string;
+  tags: string[];
+  targets: {
+    [target: string]: {
+      executor: string;
+    } & Partial<{ [other: string]: unknown }>
+  }
+}
+
+interface NxWorkspace {
+  version: 2;
+  projects: {
+    [name: string]: string | NxProject;
+  }
+}
+
+
+class NxRepositoryParser extends RepoParserBase implements RepoParser {
+  public readonly type: string = 'Nx v2 Workspace';
+
+  private useExperimentalCli?: boolean;
+
+  constructor(workspaceRoot: string, log: Log, pathToJest: string) {
+    super(workspaceRoot, log, pathToJest);
+  }
+
+  public async getConfigFileName(): Promise<string> {
+    const angularConfigFilePath = path.resolve(this.workspaceRoot, "angular.json");
+    const isAngularWorkspace = await fs.existsSync(angularConfigFilePath);
+    if (isAngularWorkspace) {
+      return "angular.json";
+    }
+    return "workspace.json";
+  }
+
+  public async getProjects(): Promise<ProjectConfig[]> {
+    const configFileName = await this.getConfigFileName();
+    await this.ensureUseExperimentalCliDetermined();
+    const buffer = await readFile(path.resolve(this.workspaceRoot, configFileName));
+    const workspaceConfig = JSON.parse(buffer.toString()) as NxWorkspace;
+
+    const projectConfigs = Object.entries(workspaceConfig.projects)
+      .map(entry => this.toNxProjectConfig(entry))
+      .filter(this.isProjectWithJestTarget)
+      .map(entry => this.toProjectConfig(entry));
+
+    return projectConfigs;
```

Comment:
We can use the `readCachedProjectGraph` function from the `nx` package to get this information. (We'd have to load this file from the workspace though: `nx/src/project-graph/project-graph`)